### ISSUE TITLE
Fails to parse file when using flow annotation for callback prop with parameter

### ISSUE THREAD

**Issue by mmsbrggr** [reactjs/react-docgen#206]
The following file cannot be parsed:
```javascript
export default class Option extends React.PureComponent {
    props: {
        selected: boolean,
        disabled: boolean,
        focus: boolean,
        value: string,
        children?: string,
        onClick?: (string) => void,
    };
    ...
}
```
I am using react-docgen indirectly through styleguidist. The following error message is output:
`Cannot parse .../Option.js: TypeError: Cannot read property 'indent' of undefined`

I guess that the issue has something to do with the `onClick` type annotation `(string) => void`, because if I change it to `(value: string) => void` it works again.

**Comment by fkling** [comment]
Do you have a stack trace?

**Comment by mmsbrggr** [comment]
@fkling:
```
Error with path "Option.js": SyntaxError: Unexpected token =>. It looks like you are trying to write a function type, but you ended up writing a grouped type followed by an =>, which is a syntax error. Remember, function type parameters are named so function types look like (name1: type1, name2: type2) => returnType. You probably wrote (type1) => returnType (1:NaN)
SyntaxError: Unexpected token =>. It looks like you are trying to write a function type, but you ended up writing a grouped type followed by an =>, which is a syntax error. Remember, function type parameters are named so function types look like (name1: type1, name2: type2) => returnType. You probably wrote (type1) => returnType (1:NaN)
    at Parser.pp.raise (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/parser/location.js:24:13)
    at Parser.pp.flowParsePrimaryType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:500:16)
    at Parser.pp.flowParsePostfixType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:557:38)
    at Parser.pp.flowParsePrefixType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:573:17)
    at Parser.pp.flowParseIntersectionType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:579:19)
    at Parser.pp.flowParseUnionType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:589:19)
    at Parser.pp.flowParseType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:600:19)
    at Parser.pp.flowParseTypeInitialiser (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:20:19)
    at Parser.pp.flowParseObjectType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:330:27)
    at Parser.pp.flowParsePrimaryType (/usr/lib/node_modules/react-docgen/node_modules/babylon/lib/plugins/flow.js:459:19)
```

**Comment by mmsbrggr** [comment]
@fkling Is there anything planed to resolve this issue?

**Comment by fkling** [comment]
You definitely have to use v3-dev, which uses the latest version of babel, otherwise the syntax is not recognized at all. However, that still won't work because react-docgen expects every argument to have a name. I will fix that.

**Comment by odinuge** [comment]
This should be solved in #218.

**Comment by fkling** [comment]
This is fixed in v3.0.0-beta8 now!
